modules/041-linstor/images/linstor-server/liveness.sh

```diff
@@ -21,6 +21,20 @@ fi
 
 # Sometimes nodes can be shown as Online without established connection to them.
 # This is a workaround for https://github.com/LINBIT/linstor-server/issues/331, https://github.com/LINBIT/linstor-server/issues/219
+
+# Collect list of satellite nodes
+SATELLITES_ONLINE=$(linstor -m --output-version=v1 n l | jq -r '.[][] | select(.type == "SATELLITE" and .connection_status == "ONLINE").name' || true)
+if [ -z "$SATELLITES_ONLINE" ]; then
+  exit 0
+fi
+
+# Check online nodes with lost connection
+linstor -m --output-version=v1 sp l -s DfltDisklessStorPool -n $SATELLITES_ONLINE | jq '.[][].reports[]?.message' | grep 'No active connection to satellite'
+if [ $? -eq 0 ]; then
+  exit 1
+fi
```

Comment:
```suggestion
```suggestion
if linstor -m --output-version=v1 storagepool list -s DfltDisklessStorPool -n $SATELLITES_ONLINE | jq '.[][].reports[]?.message' | grep -q 'No active connection to satellite'; then
  exit 1
fi
``````